src/aws-cpp-sdk-core/source/auth/signer/AWSAuthV4Signer.cpp

```diff
@@ -238,7 +238,13 @@ bool AWSAuthV4Signer::SignRequest(Aws::Http::HttpRequest& request, const char* r
         if (request.GetRequestHash().second != nullptr)
         {
             Aws::String checksumHeaderKey = Aws::String("x-amz-checksum-") + request.GetRequestHash().first;
-            Aws::String checksumHeaderValue = HashingUtils::Base64Encode(request.GetRequestHash().second->Calculate(*(request.GetContentBody())).GetResult());
+            auto checksumHeaderValue = [](const std::shared_ptr<Utils::Crypto::Hash> &hash,
```

Comment:
I know IIFE is cool, but KISS is also cool and ternary operator would work here too:

```
auto checksumHeaderValue = request.GetRequestHash().second ? 
    request.GetRequestHash().second : HashingUtils::Base64Encode(hash->Calculate(request.GetContentBody().get()).GetResult());
```

also, because we now dereference GetContentBody, it worth adding a check it is present (and trace error and return false if not)

Reply:
or keep the IIFE, resulting ternary operator is also looking complicated.